demos/common/utilities/jws_using_mbedtls.c

```diff
@@ -0,0 +1,634 @@
+/* Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT License. */
+
+#include "jws.h"
+
+#include "azure/az_core.h"
+#include "azure/az_iot.h"
+
+#include "azure_iot_result.h"
+#include "azure_iot_json_reader.h"
+#include "azure_iot_adu_client.h"
+
+#include "mbedtls/base64.h"
+#include "mbedtls/rsa.h"
+#include "mbedtls/pk.h"
+#include "mbedtls/ctr_drbg.h"
+#include "mbedtls/entropy.h"
+#include "mbedtls/cipher.h"
+
+/* For logging */
+#include "demo_config.h"
+
+/**
+ * @brief Convenience macro to return if an operation failed.
+ */
+#define azureiotresultRETURN_IF_FAILED( exp )        \
+    do                                               \
+    {                                                \
+        AzureIoTResult_t const _xAzResult = ( exp ); \
+        if( _xAzResult != eAzureIoTSuccess )         \
+        {                                            \
+            return _xAzResult;                       \
+        }                                            \
+    } while( 0 )
+
+#define jwsRSA3072_SIZE            384
+#define jwsSHA256_SIZE             32
+#define jwsPKCS7_PAYLOAD_OFFSET    19
+
+#define jwsSHA256_JSON_VALUE       "sha256"
+#define jwsSJWK_JSON_VALUE         "sjwk"
+#define jwsKID_JSON_VALUE          "kid"
+#define jwsN_JSON_VALUE            "n"
+#define jwsE_JSON_VALUE            "e"
+#define jwsALG_JSON_VALUE          "alg"
+
+static unsigned char ucJWSHeader[ 1400 ];
```

Comment:
Where does the number 1400 come from?  And these #'s generally?  Could we link to a spec someplace with like limits or JWS spec or ...?

Reply:
This is purely based on what I observed with the manifest that was incoming. This isn't a spec dictated thing, but comes from how ADU composes the nesting doll situation.